server/sonar-server/src/main/java/org/sonar/server/permission/ws/UpdateTemplateAction.java

```diff
@@ -0,0 +1,129 @@
+    this.dbClient = dbClient;
+    this.userSession = userSession;
+    this.system = system;
+    this.finder = finder;
+  }
+
+  @Override
+  public void define(WebService.NewController context) {
+    WebService.NewAction action = context.createAction("update_template")
+      .setDescription("Update a permission template.<br />" +
+        "It requires administration permissions to access.")
+      .setResponseExample(getClass().getResource("update_template-example.json"))
+      .setSince("5.2")
+      .setPost(true)
+      .setHandler(this);
+
+    createTemplateKeyParameter(action);
+
+    action.createParam(PARAM_TEMPLATE_NAME)
+      .setDescription("Name")
+      .setExampleValue("Financial Service Permissions");
+
+    createTemplateProjectKeyPatternParameter(action);
+    createTemplateDescriptionParameter(action);
+  }
+
+  @Override
+  public void handle(Request wsRequest, Response wsResponse) throws Exception {
+    checkGlobalAdminUser(userSession);
+
+    String key = wsRequest.mandatoryParam(PARAM_TEMPLATE_KEY);
+    String nameParam = wsRequest.param(PARAM_TEMPLATE_NAME);
+    String descriptionParam = wsRequest.param(PARAM_TEMPLATE_DESCRIPTION);
+    String projectPatternParam = wsRequest.param(PARAM_TEMPLATE_PATTERN);
+
+    DbSession dbSession = dbClient.openSession(false);
+    try {
+      PermissionTemplateDto templateToUpdate = getAndBuildTemplateToUpdate(dbSession, key, nameParam, descriptionParam, projectPatternParam);
+      validateTemplate(dbSession, templateToUpdate);
+      PermissionTemplateDto updatedTemplate = updateTemplate(dbSession, templateToUpdate);
+
+      UpdatePermissionTemplateResponse response = buildResponse(updatedTemplate);
+      writeProtobuf(response, wsRequest, wsResponse);
+    } finally {
+      dbClient.closeSession(dbSession);
+    }
+  }
+
+  private void validateTemplate(DbSession dbSession, PermissionTemplateDto templateToUpdate) {
+    validateTemplateNameForUpdate(dbSession, dbClient, templateToUpdate.getName(), templateToUpdate.getId());
+    validateProjectPattern(templateToUpdate.getKeyPattern());
+  }
+
+  private PermissionTemplateDto getAndBuildTemplateToUpdate(DbSession dbSession, String key, @Nullable String newName, @Nullable String newDescription,
+    @Nullable String newProjectKeyPattern) {
+    PermissionTemplateDto templateToUpdate = finder.getTemplate(dbSession, key);
+    templateToUpdate.setName(firstNonNull(newName, templateToUpdate.getName()));
+    templateToUpdate.setDescription(firstNonNull(newDescription, templateToUpdate.getDescription()));
+    templateToUpdate.setKeyPattern(firstNonNull(newProjectKeyPattern, templateToUpdate.getKeyPattern()));
+    templateToUpdate.setUpdatedAt(new Date(system.now()));
+
+    return templateToUpdate;
+  }
+
+  private PermissionTemplateDto updateTemplate(DbSession dbSession, PermissionTemplateDto templateToUpdate) {
+    return dbClient.permissionTemplateDao().update(dbSession, templateToUpdate);
```

Comment:
because the commit is done in the DAO and shared with other methods. I'll add the commit when I'll remove other uses of the method